provider/azure/environ_network.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package azure
+
+import (
+	stdcontext "context"
+
+	azurenetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-08-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+)
+
+var _ environs.NetworkingEnviron = &azureEnviron{}
+
+// SupportsSpaces implements environs.NetworkingEnviron.
+func (env *azureEnviron) SupportsSpaces(context.ProviderCallContext) (bool, error) {
+	return false, nil
+}
+
+// Subnets implements environs.NetworkingEnviron.
+func (env *azureEnviron) Subnets(
+	_ context.ProviderCallContext, instanceID instance.Id, _ []network.Id) ([]network.SubnetInfo, error) {
+	if instanceID != instance.UnknownId {
+		return nil, errors.NotSupportedf("subnets for instance")
```

Comment:
There are currently no invocations of this method that supply an actual instance ID.